helm-app-operator/pkg/helm/engine.go

```diff
@@ -49,24 +52,40 @@ func (o *OwnerRefEngine) Render(chart *chart.Chart, values chartutil.Values) (ma
 }
 
 // addOwnerRefs adds the configured ownerRefs to a single rendered file
+// Adds the ownerrefs to all the documents in a YAML file
 func (o *OwnerRefEngine) addOwnerRefs(fileContents string) (string, error) {
-	parsed := chartutil.FromYaml(fileContents)
-	if errors, ok := parsed["Error"]; ok {
-		return "", fmt.Errorf("error parsing rendered template to add ownerrefs: %v", errors)
-	}
+	const documentSeparator = "---\n"
 
-	// Empty input
-	if len(parsed) == 0 {
-		return "", nil
-	}
+	var outBuf bytes.Buffer
+	manifests := releaseutil.SplitManifests(fileContents)
 
-	unst, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&parsed)
-	if err != nil {
-		return "", err
+	for _, manifest := range manifests {
+		manifestMap := chartutil.FromYaml(manifest)
+		if errors, ok := manifestMap["Error"]; ok {
+			return "", fmt.Errorf("error parsing rendered template to add ownerrefs: %v", errors)
+		}
+
+		// Check if the document is empty
+		if len(manifestMap) == 0 {
+			continue
+		}
+
+		unst, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&manifestMap)
+		if err != nil {
+			return "", err
+		}
+
+		unstructured := &unstructured.Unstructured{Object: unst}
+		unstructured.SetOwnerReferences(o.refs)
+
+		// Write the document with owner ref to the buffer
+		_, err = outBuf.WriteString(chartutil.ToYaml(unstructured.Object) + documentSeparator)
+		if err != nil {
+			return "", fmt.Errorf("error writing the document to buffer : %v", err)
```

Comment:
Nitpick, but could you remove the space between buffer and the colon (i.e `buffer :` -> `buffer:`).

Reply:
Refactored